lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/helpers/AuthorizationRequestHelper.java

```diff
@@ -85,24 +83,7 @@ ResponseType.CODE, new ClientID(oauthCriConfig.getClientId()))
                         .claim("govuk_signin_journey_id", govukSigninJourneyId);
 
         if (Objects.nonNull(sharedClaims)) {
-            if (sharedClaims.getEmailAddress() != null) {
-                claimsSetBuilder.claim(SHARED_CLAIMS, sharedClaims);
-            } else if (!sharedClaims.getSocialSecurityRecord().isEmpty()) {
-                NinoSharedClaimsResponseDto response =
-                        new NinoSharedClaimsResponseDto(
-                                sharedClaims.getName(),
-                                sharedClaims.getBirthDate(),
-                                sharedClaims.getAddress(),
-                                sharedClaims.getSocialSecurityRecord());
-                claimsSetBuilder.claim(SHARED_CLAIMS, response);
-            } else {
-                SharedClaimsResponseDto response =
-                        new SharedClaimsResponseDto(
-                                sharedClaims.getName(),
-                                sharedClaims.getBirthDate(),
-                                sharedClaims.getAddress());
-                claimsSetBuilder.claim(SHARED_CLAIMS, response);
-            }
+            claimsSetBuilder.claim(SHARED_CLAIMS, sharedClaims.toMapWithNoNulls());
```

Comment:
By default Nimbus does its own JSON serialisation which ignores our Jackson annotations and include null claims.  Using Jackson to go via a map will exclude them - as with the evidence request below.